test/onnx/pytorch_test_common.py

```diff
@@ -301,7 +283,29 @@ def xfail(reason: str):
     Returns:
         A decorator for expecting test failure.
     """
-    return unittest.expectedFailure
+
+    def wrapper(func):
+        @functools.wraps(func)
+        def inner(self, *args, **kwargs):
+            try:
+                func(self, *args, **kwargs)
+            except Exception as e:
+                if isinstance(e, torch.onnx.OnnxExporterError):
+                    # diagnostic message is in the cause of the exception
+                    assert error_message in str(
+                        e.__cause__
+                    ), f"Expected error message: {error_message} NOT in {str(e.__cause__)}"
```

Comment:
Consider using regex for this? Example api: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex